### PR TITLE
Cleaned fittingmetric new/delete

### PR DESCRIPTION
**Pull request by philipmnel**
## Description
This PR replaces `new`/`delete` in `lib3index/fittingmetric.cc` with `std::vector` per task in #2412  

## Checklist
- [x] Quick tests pass.

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

File: psi4/src/psi4/lib3index/fittingmetric.cc [hunk 1]
```diff
@@ -387,13 +380,12 @@ void FittingMetric::form_QR_inverse(double tol) {
...
         // First, find out how much workspace to provide
-        C_DORGQR(n, n, n, Qp[0], n, tau, &work_size, -1);
+        C_DORGQR(n, n, n, Qp[0], n, tau.data(), &work_size, -1);
 
         // Now, form Q
         lwork = (int)work_size;
-        work = new double[lwork];
-        C_DORGQR(n, n, n, Qp[0], n, tau, work, lwork);
-        delete[] work;
+        work.resize(lwork);
```
Comment (JonathonMisiewicz) [inline psi4/src/psi4/lib3index/fittingmetric.cc:387]: `work` is already of size `lwork`, so this line isn't necessary.

**Comment by JonathonMisiewicz** [comment]
Thanks for taking this on! There's one unnecessary line of code here, but after that, this looks good to me.

File: psi4/src/psi4/lib3index/fittingmetric.cc (hunk 1 above)
Comment (philipmnel) [reply psi4/src/psi4/lib3index/fittingmetric.cc:387]: `lwork` gets resized from `C_DORGQR(n, n, n, Qp[0], n, tau.data(), &work_size, -1);` if I'm not mistaken.

Comment (JonathonMisiewicz) [reply psi4/src/psi4/lib3index/fittingmetric.cc:387]: That's right. In that case, please update the comment above the line you quoted to explicitly say it writes the optimal length into `work_size`. I would not have guessed that without checking to see what exactly that function does.

**Review by JonathonMisiewicz — APPROVED** [review]

**Review by jturney — APPROVED** [review]
Nice cleanup!

**Review by zachglick — APPROVED** [review]
LGTM!

**Pull request merged**